packages/seo/src/lib/seo.service.ts

```diff
@@ -50,6 +52,14 @@ export class SeoService {
     ]);
   }
 
+  public setCanonical(url: string): void {
+    this.generateTags([{ name: 'canonical', content: url }]);
```

Comment:
I dont think this will work this way since `canonical` tag is a `link` tag and `this.generateTags` create `meta` tags

Maybe we can just use `document` directly in this case (`document.head.appendChild`)?

Reply:
You are right, but using document would conflict when trying to use it with ssr, no?

Reply:
we can inject the angular provided wrapper of the `document` element which works with ssr

Reply:
But I just realized that setUrl already does it, I think this is no longer necessary.